piker/clearing/_paper_engine.py

```diff
@@ -271,6 +273,24 @@ async def fake_fill(
             dt=pendulum.from_timestamp(fill_time_s),
             bsuid=key,
         )
+
+        # Update in memory ledger per trade
+        ledger_entry = {oid: t.to_dict()}
+
+        # Store txn in state for PP update
+        self._txn_dict[oid] = t
+        self._trade_ledger.update(ledger_entry)
+
+        # Write to ledger toml right now
+        with open_trade_ledger(self.broker, 'paper') as ledger:
+            ledger.update(self._trade_ledger)
+
+        # Write to pps toml right now
+        with open_pps(self.broker, 'piker-paper') as table: 
+            table.update_from_trans(self._txn_dict)
+            # save pps in local state
+            self._positions.update(table.pps)
+
         pp.add_clear(t)
 
         pp_msg = BrokerdPosition(
```

Comment:
Ok so we have a more major issue here because this `pp_msg` isn't being updated including the state of the existing positions. This is likely the source of the second issue described [in this comment](https://github.com/pikers/piker/pull/462#issuecomment-1438852266).

The `BrokerdPosition` msg needs to be generated inclusive of the existing state held by `table.pps`; as it stands right now this will always just send the newly generated `pp: Position` value(s) instead of the instance: `table.pps[fqsn]` *i think*.

Reply:
Further, this **can for sure** be caught by the tests, you just need to do an assertion on the msgs received on the test's handle to the `trades_stream: tractor.MsgStream` delivered by,
```python
            open_ems(fqsn, mode="paper") as (
                book,
                trades_stream,
                pps,
                accounts,
                dialogs,
            ),
```

So ideally you iterate this stream as part of the test, and wait until you get all position msgs expected as in the `pps.toml`.